Sources/DistributedActors/Pattern/WorkerPool.swift

```diff
@@ -44,19 +54,79 @@ public struct WorkerPool<Message: ActorMessage> {
         /// The worker pool will terminate itself if all of its static workers have terminated.
         /// You may death-watch the worker pool in order to react to this situation, e.g. by spawning a replacement pool,
         /// or gracefully shutting down your application.
-        case `static`([_ActorRef<Message>])
+        case `static`([Worker])
     }
 
     // ==== ------------------------------------------------------------------------------------------------------------
     // MARK: State shared across all states of the WorkerPool
 
-    var settings: WorkerPoolSettings<Message>
+    var settings: WorkerPoolSettings<Worker>
     var selector: Selector {
         self.settings.selector
     }
```

Comment:
So since we store the selector, we'd keep the `.static[Worker]` as well, making them not able to terminate...

Maybe we should not store `Selector` but some `_Selector` that is same as Selector but for `static` is just `case static` rather than keeping the actors retained 🤔

Reply:
The same goes for `WorkerPoolSettings` too? I got rid of the `settings` and `selector` properties, and instead store only `logLevel` and `whenAllWorkersTerminated`, which doesn't have any `Worker`.

Reply:
Amended https://github.com/apple/swift-distributed-actors/pull/912/commits/0ed16f138acdfabc6e7891250a385e11b579c4bf

Reply:
Right makes sense... since now storing "refs" (actors) actually forces their extended lifetime, so we can't du stuff we did with refs.